packages/ui/src/components/form-input/components/form-text-input.tsx

```diff
@@ -36,13 +22,10 @@ const FormTextInput = forwardRef<HTMLInputElement, FormTextInputPropsType>((prop
       name={props.name}
       control={formContext.control}
       render={({ field, fieldState }) => {
-        // Use proper React Hook Form reference handling
-        field.ref = setRefs
-
         // form error takes precedence over props.error
-        const errorMessage = fieldState.error?.message || props.error
+        const error = fieldState.error?.message || props.error
 
-        return <TextInput {...props} {...field} error={errorMessage} />
+        return <TextInput {...props} {...field} error={error} ref={ref} theme={error ? 'danger' : 'default'} />
```

Comment:
No need to pass in the `theme` since it's determined from in the `TextInput` component.
```suggestion
        return <TextInput {...props} {...field} error={fieldState.error?.message || props.error} ref={ref} />
```

Reply:
Sorry, updated